.github/workflows/flow-micro-benchmarks.yml

```diff
@@ -0,0 +1,122 @@
+name: Run a Micro Benchmark Flow
+
+on:
+    # Manual trigger
+    workflow_dispatch:
+      inputs:
+        profile_env:
+          type: number # for default of 0
+        rejson_branch:
+          type: string
+          default: master
+          description: 'branch to use for rejson'
+        benchmark_folder: 
+          type: string
+          default: src/redisearch_rs/trie_bencher
+          description: folder containing the benchmark to be run with `cargo bench`
+        output_folder:
+          type: string
+          default: bin/criterion/trie
+          description: folder used to store the output of criterion
+        memory_limit:
+          type: number
+          default: 12582912 # that's 12GB
+          description: Used by the memory monitoring to understand if a runner terminates because of lack of memory (leaking)
+    
+    # For calling from other workflows
+    workflow_call:
+      inputs:
+        profile_env:
+          type: number # for default of 0
+        rejson_branch:
+          type: string
+          default: master
+          description: 'branch to use for rejson'
+        benchmark_folder: 
+          type: string
+          default: src/redisearch_rs/trie_bencher
+          description: folder containing the benchmark to be run with `cargo bench`
+        output_folder:
+          type: string
+          default: bin/criterion/trie
+          description: folder used to store the output of criterion
+        memory_limit:
+          type: number
+          default: 12582912 # that's 12GB
+          description: Used by the memory monitoring to understand if a runner terminates because of lack of memory (leaking)
+
+jobs:
+  benchmarks:
+    runs-on: ubuntu-24.04
+    env:
+      RUST_BACKTRACE: full
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: Setup specific  # assuming node20 supported
+        working-directory: .install
+        run: |
+          ./install_script.sh ${{ steps.mode.outputs.mode }} sudo
+      - name: Setup tests dependencies
+        run: |
+          .install/test_deps/common_installations.sh ${{ steps.mode.outputs.mode }} sudo
+      - name: Install Boost
+        working-directory: .install
+        run: ./install_boost.sh ${{ env.BOOST_VERSION }} sudo
+
+      - name: Build RediSearch
+        run: make build
+            PROFILE=${{ inputs.profile_env }}
+            SHOW=1
+      - name: Prepare automation
+        run: |
+          ./deps/readies/bin/getpy3
+          pip3 install --upgrade pip
+          python3 -m pip install -r tests/benchmarks/requirements.txt
+          VERSION=1.11.1 sudo ./deps/readies/bin/getterraform
+      - name: Prepare ReJSON Module
+        run: REJSON_BRANCH=${{ inputs.rejson_branch }} ./tests/deps/setup_rejson.sh
+      
```

Comment:
This setup feels overly complicated. We should be able to get away with something as simple as what we have in `task-lint.yml`.